packages/api/src/routes/internal/integration/quest.ts

```diff
@@ -72,11 +94,15 @@ router.get(
  * @returns 200 OK
  */
 router.post(
-  "/upload-roster",
+  "/upload-roster/:rosterType",
   requestLogger,
-  asyncHandler(async (_: Request, res: Response) => {
+  asyncHandler(async (req: Request, res: Response) => {
+    const rosterType = getFromParamsOrFail("rosterType", req);
+    if (!isRosterType(rosterType)) {
+      return res.sendStatus(status.BAD_REQUEST);
+    }
```

Comment:
NABD, but it's nice to use zod schemas for these kind of things as
1. we need to define types elsewhere anyways, so minimize the amount of code we need to write; and
2. it provides a better response when the data doesn't match the endpoint's requirements.